src/cpp/shared_core/include/shared_core/json/Json.hpp

```diff
@@ -392,6 +392,18 @@ class Value
     */
    Error validate(const std::string& in_schema) const;
 
+   /**
+    * @brief Attempts to coerce a JSON object to conform to the given schema by discarding non-conforming
+    *    properties.
+    *
+    * @param in_schema           The schema to validate this value against.
+    * @param out_propViolations  The names of of the properties that did not conform to the schema.
+    *
+    * @return Success if this JSON value matches the schema after coercion; Error otherwise.
+    */
+   Error coerce(const std::string& in_schema,
```

Comment:
Nit: Could you move this definition below `clone()` and above `getArray()`?

Reply:
Will do.